drivers/ethernet/phy/phy_qualcomm_ar8031.c

```diff
@@ -0,0 +1,526 @@
+#define PHY_SPEED_100M					(1U << 14)
+#define PHY_SPEED_1000M					(2U << 14)
+
+/* The PHY Debug port address register */
+#define PHY_DEBUGPORT_ADDR_REG				0x1DU
+/* The PHY Debug port data register */
+#define PHY_DEBUGPORT_DATA_REG				0x1EU
+
+ /* Debug port registers */
+/* Analog Test Control */
+#define PHY_DEBUGPORT_ANALOG_CTRL			0x0
+#define PHY_DEBUGPORT_ANALOG_CTRL_RX_DELAY		(1U << 15)
+/* SerDes Test and System Mode Control */
+#define PHY_DEBUGPORT_SD_SM_CTRL			0x5
+#define PHY_DEBUGPORT_SD_SM_CTRL_TX_DELAY		(1U << 8)
+
+/* PHY MMD device address 3 */
+#define PHY_MMD_DEVICEID3				3U
+/* EEE capability Register */
+#define PHY_MMD3_EEE_CAP				0x14U
+/* MMD3: smartEEE control 3 Register */
+#define PHY_MMD3_SMARTEEE_CTRL3				0x805DU
+#define PHY_MMD3_SMARTEEE_CTRL3_LPI_EN			(1U << 8)
+
+/* PHY MMD device address 7 */
+#define PHY_MMD_DEVICEID7				7U
+/* EEE advertisement Register */
+#define PHY_MMD7_AN_EEE_ADV				0x3CU
+ /* EEE mask common for adv and cap */
+#define PHY_EEE_1000T					0x4U
+#define PHY_EEE_100TX					0x2U
+
+struct qc_ar8031_config {
+	uint8_t addr;
+	bool no_reset;
+	bool fixed_link;
+	bool enable_eee;
+	int fixed_speed;
+	const struct device *mdio_dev;
+};
+
+struct qc_ar8031_data {
+	const struct device *dev;
+	phy_callback_t cb;
+	void *cb_data;
+	struct k_work_delayable monitor_work;
+	struct phy_link_state state;
+	struct k_sem sem;
+};
+
+static int qc_ar8031_get_link_state(const struct device *dev,
+				  struct phy_link_state *state);
+
+static int qc_ar8031_read(const struct device *dev,
+				uint16_t reg_addr, uint32_t *data)
+{
+	const struct qc_ar8031_config *config = dev->config;
+	int ret;
+
+	/* Make sure excessive bits 16-31 are reset */
+	*data = 0U;
+
+	/* Read the PHY register */
+	ret = mdio_read(config->mdio_dev, config->addr, reg_addr, (uint16_t *)data);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int qc_ar8031_write(const struct device *dev,
+				uint16_t reg_addr, uint32_t data)
+{
+	const struct qc_ar8031_config *config = dev->config;
+	int ret;
+
+	ret = mdio_write(config->mdio_dev, config->addr, reg_addr, (uint16_t)data);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int qc_ar8031_mmd_set_device(const struct device *dev, uint32_t device,
+				uint32_t addr, uint16_t mode)
+{
+	uint16_t reg_value = (device & MII_MMD_ACR_DEVAD_MASK) | MII_MMD_ACR_ADDR;
+
+	if (qc_ar8031_write(dev, MII_MMD_ACR, reg_value) < 0) {
+		return -EIO;
+	}
+	if (qc_ar8031_write(dev, MII_MMD_AADR, addr) < 0) {
+		return -EIO;
+	}
+	/* Set Function mode of data access(b01~11) and device address. */
+	if (qc_ar8031_write(dev, MII_MMD_ACR, (device & MII_MMD_ACR_DEVAD_MASK) | mode) < 0) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int qc_ar8031_mmd_read(const struct device *dev, uint32_t device,
+				uint32_t addr, uint32_t *data)
+{
+	int ret;
+
+	*data = 0U;
+
+	ret = qc_ar8031_mmd_set_device(dev, device, addr, MII_MMD_ACR_DATA_NO_POS_INC);
+	if (ret)
+		return -EIO;
```

Comment:
I thought this was a compliance check, please always use curly braces.

```suggestion
	if (ret) {
		return -EIO;
	}
```

Reply:
> I thought this was a compliance check, please always use curly braces.

fixed, thanks.